examples/object_detector/detectron2/detectron2-handler.py

```diff
@@ -0,0 +1,265 @@
+        """
+        Perform inference on the model input.
+
+        Args:
+            model_input (List[np.ndarray]): List of preprocessed images.
+
+        Returns:
+            List[Dict]: List of inference outputs.
+        """
+        logger.info(f"Inference started for a batch of {len(model_input)}.")
+
+        outputs = []
+        for idx, image in enumerate(model_input):
+            try:
+                logger.debug(f"Processing image {idx}: shape={image.shape}, dtype={image.dtype}")
+                output = self.predictor(image)
+                outputs.append(output)
+            except Exception as e:
+                logger.exception(f"Error during inference on image {idx}")
+                raise e
+        logger.info(f"Inference finished for a batch of {len(model_input)}.")
+        return outputs
+
+    def postprocess(self, inference_outputs):
+        """
+        Post-process the inference outputs to a serializable format.
+
+        Args:
+            inference_outputs (List[Dict]): List of inference outputs.
+
+        Returns:
+            List[str]: List of JSON strings containing predictions.
+        """
+        start_time = time.time()
+        logger.info(f"Post-processing started at {start_time} for a batch of {len(inference_outputs)}.")
+        responses = []
+        for idx, output in enumerate(inference_outputs):
+            try:
+                predictions = output["instances"].to("cpu")
+                logger.debug(f"Available prediction fields: {predictions.get_fields().keys()}")
+                response = {}
+                if predictions.has("pred_classes"):
+                    classes = predictions.pred_classes.numpy().tolist()
+                    response["classes"] = classes
+                if predictions.has("pred_boxes"):
+                    boxes = predictions.pred_boxes.tensor.numpy().tolist()
+                    response["boxes"] = boxes
+                if predictions.has("scores"):
+                    scores = predictions.scores.numpy().tolist()
+                    response["scores"] = scores
+                if predictions.has("pred_masks"):
+                    response["masks_present"] = True
+                responses.append(json.dumps(response))
+            except Exception as e:
+                logger.exception(f"Error during post-processing of output {idx}")
+                raise e
+        elapsed_time = time.time() - start_time
+        logger.info(f"Post-processing finished for a batch of {len(inference_outputs)} in {elapsed_time:.2f} seconds.")
+
+        return responses
+
+    def handle(self, data, context):
+        """
+        Entry point for TorchServe to interact with the ModelHandler.
+
+        Args:
+            data (List[Dict]): Input data.
+            context (Context): Model server context.
+
+        Returns:
+            List[str]: List of predictions.
+        """
+        logger.info("Handling request...")
+        start_time = time.time()
```

Comment:
We already have a utility to measure the time, please check this handler 
https://github.com/pytorch/serve/blob/0985386d5a9bc9dd4fe0feaa7af186e2e5b3a84b/examples/pt2/torch_compile_openvino/stable_diffusion/stable_diffusion_handler.py#L96